File: BlogServer/Blog.Model/Param/RandomWriteParam.cs
using System.ComponentModel;

namespace Blog.Model.Param
{
    public class RandomWriteInsertParam
    {
        public string[]? Cover_list {  get; set; }
        public int CreateUserId { get; set; }
        public required string Content {  get; set; }
    }

    public class RandomWriteUpdateParam: RandomWriteInsertParam
    {
        public required int Id { get; set; }
        public new int? CreateUserId { get; set; } = null;
    }

    public class RandomWriteFindParam : PageParam
    {
        [DefaultValue(null)]
        public string? Content { get; set; } = null;
    }
}

File: blog-server/src/module/admin/randomWrite/model/RandomWrite.param.ts
    import {IsString, IsInt, IsArray, IsOptional} from 'class-validator'
    import {PageParam} from "../../../../model/BaseParam";


export class RandomWriteInsertParam {

    @IsString()
    content: string

    @IsArray()
    @IsOptional()       // 表示 search 字段是可选的
    cover_list?: string[]

    @IsInt()
    @IsOptional()
    CreateUserId?: number
}


export class RandomWriteUpdateParam extends RandomWriteInsertParam{
    @IsInt({ message: 'Id必须是整数类型' })
    id: number;

}

export class RandomWriteFindParam extends PageParam{
    @IsString()
    @IsOptional()       // 表示 search 字段是可选的
    content?: string
}
